### PR TITLE
feat(ui): convert last bits to react

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: src/traceViewer/web/ui/sourceTab.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import * as monaco from 'monaco-editor';
+import { ActionEntry } from '../../traceModel';
+import * as React from 'react';
+import { useAsyncMemo, useMeasure } from './helpers';
+
+export const SourceTab: React.FunctionComponent<{
+  actionEntry: ActionEntry | undefined,
+}> = ({ actionEntry }) => {
+  const location = useAsyncMemo<{ fileName?: string, lineNumber?: number, value?: string }>(async () => {
```
Comment (JoelEinbinder) [inline src/traceViewer/web/ui/sourceTab.tsx:25]: This is sync? Same as below. useMemo?

Comment (JoelEinbinder) [inline src/traceViewer/web/ui/sourceTab.tsx:25]:
```suggestion
  const location = React.useMemo<{ fileName?: string, lineNumber?: number, value?: string }>(() => {
```

File: src/traceViewer/web/ui/sourceTab.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      return { value: action.stack! };
+    const match = frame.match(/at [^(]+\(([^:]+):(\d+):\d+\)/) || frame.match(/at ([^:^(]+):(\d+):\d+/);
+    if (!match)
+      return { value: action.stack! };
+    const fileName = match[1];
+    const lineNumber = parseInt(match[2], 10);
+    return { fileName, lineNumber };
+  }, [actionEntry], {});
+
+  const content = useAsyncMemo<string>(async () => {
```
Comment (JoelEinbinder) [inline src/traceViewer/web/ui/sourceTab.tsx:41]:
```suggestion
  const content = React.useMemo<string>(() => {
```

File: src/traceViewer/web/ui/sourceTab.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      editor.editor.layout();
+  }, [editor, measure.width, measure.height]);
+
+  return <div ref={ref} style={{ flex: 'auto', minWidth: '0', minHeight: '0' }}></div>;
+};
+
+
+function decorateLine(editor: monaco.editor.IStandaloneCodeEditor, lineNumber: number | undefined) {
+  if (lineNumber !== undefined) {
+    editor.deltaDecorations([], [{
```
Comment (JoelEinbinder) [inline src/traceViewer/web/ui/sourceTab.tsx:88]: Who cleans up this decoration? I thought you are supposed to pass the old decorations in as the first argument here.

**Review by JoelEinbinder — APPROVED** [review]

File: src/traceViewer/web/ui/sourceTab.tsx (hunk 3 above)
Comment (dgozman) [reply src/traceViewer/web/ui/sourceTab.tsx]: Fixed.

**Pull request merged**